Packs/AggregatedScripts/Scripts/IndicatorEnrichment/IndicatorEnrichment.py

```diff
@@ -0,0 +1,530 @@
+             - Over 100 indicators given (after clearing invalid and duplicates)
+            GracefulExit: If text input yields no valid indicators (non-error state).
+        """
+        total_valid = sum(len(v) for v in self.valid_indicators_by_type.values())
+        has_invalid = bool(self.unsupported_items or self.unknown_items)
+
+        text_provided = bool(self.raw_arguments.get("text", "").strip())
+        list_key_present = "indicator_list" in self.raw_arguments
+        list_has_values = bool(argToList(self.raw_arguments.get("indicator_list")))
+
+        if not text_provided and not list_has_values and not list_key_present:
+            raise ValidationError(MSG_NO_VALID_INDICATORS)
+
+        if list_key_present and (not list_has_values or total_valid == 0):
+            raise ValidationError(MSG_NO_VALID_INDICATORS)
+
+        if text_provided and not list_key_present and total_valid == 0 and not has_invalid:
+            raise GracefulExit(MSG_NO_VALID_INDICATORS)
+
+        ignore_limit = argToBoolean(self.raw_arguments.get("ignore_indicator_limit", False))
+        if not ignore_limit and total_valid > MAX_INDICATORS:
+            raise ValidationError(ERR_LIMIT_TEMPLATE.format(found=total_valid, limit=MAX_INDICATORS))
+
+    def _filter_passthrough_arguments(self) -> dict[str, Any]:
+        """
+        Filters the raw arguments to include only those intended for child commands.
+
+        Returns:
+            A dictionary of arguments to pass to the enrichment commands.
+        """
+        keys = ["external_enrichment", "brands", "additional_fields"]
+        return {k: self.raw_arguments.get(k) for k in keys if self.raw_arguments.get(k) is not None}
+
+
+class EnrichmentService:
+    """
+    Coordinates the execution of child enrichment commands based on the request.
+    """
+
+    def __init__(self):
+        self._result = EnrichmentResult()
+
+    def execute(self, request: EnrichmentRequest) -> EnrichmentResult:
+        """
+        Executes the enrichment plan.
+
+        Args:
+            request: The EnrichmentRequest containing grouped indicators.
+
+        Returns:
+            An EnrichmentResult object containing aggregated outputs.
+        """
+
+        if not request.valid_indicators_by_type:
+            return self._result
+
+        tasks = self._create_execution_plan(request)
+
+        batch_output = BatchExecutor().execute_batch([t.command for t in tasks], brands_to_run=None, verbose=False)
+
+        for task, output in zip(tasks, batch_output):
+            self._parse_task_output(task, output)
+
+        return self._result
+
+    def _create_execution_plan(self, request: EnrichmentRequest) -> list[EnrichmentTask]:
+        """Creates a list of tasks mapping indicator types to configured Commands."""
+        tasks = []
+        for type_enum, indicators in request.valid_indicators_by_type.items():
+            cmd = self._build_command(type_enum, indicators, request.sub_command_arguments)
```

Comment:
does request.sub_command_arguments need to be passed? isnt this already on the self?